File: tools/sdks/go/examples/quickstart_list.go
//go:build ignore

// quickstart_list — List all PATs via GET /v1/pats.
//
// Customer concept: server-side authoritative credential inventory.
// Plaintext tokens are never returned (only metadata + last-4 prefix).
//
// Run: CORELINK_PAT=$PAT go run examples/quickstart_list.go
package main

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io"
	"net/http"
	"os"
	"time"
)

func main() {
	api := os.Getenv("CORELINK_API_URL")
	if api == "" {
		api = "https://sandbox.corelink.dev"
	}
	pat := os.Getenv("CORELINK_PAT")
	if pat == "" {
		fmt.Fprintln(os.Stderr, "error: CORELINK_PAT env var is required")
		os.Exit(2)
	}
	req, err := http.NewRequest("GET", api+"/v1/pats", nil)
	if err != nil {
		fmt.Fprintf(os.Stderr, "build request: %v\n", err)
		os.Exit(1)
	}
	req.Header.Set("Authorization", "Bearer "+pat)
	req.Header.Set("Accept", "application/json")
	client := &http.Client{Timeout: 10 * time.Second}
	resp, err := client.Do(req)
	if err != nil {
		fmt.Fprintf(os.Stderr, "http: %v\n", err)
		os.Exit(1)
	}
	defer resp.Body.Close()
	raw, _ := io.ReadAll(resp.Body)
	if resp.StatusCode/100 != 2 {
		fmt.Fprintf(os.Stderr, "server returned %d: %s\n", resp.StatusCode, string(raw))
		os.Exit(1)
	}
	var decoded map[string]any
	count := 0
	if err := json.Unmarshal(raw, &decoded); err == nil {
		if items, ok := decoded["items"].([]any); ok {
			count = len(items)
		}
	}
	var pretty bytes.Buffer
	_ = json.Indent(&pretty, raw, "", "  ")
	fmt.Printf("status=%d pat_count=%d\n%s\n", resp.StatusCode, count, pretty.String())
}

File: tools/sdks/python/examples/quickstart_list.py
#!/usr/bin/env python3
"""quickstart_list — List all PATs via GET /v1/pats.

Customer concept: server-side authoritative credential inventory.
Plaintext tokens are never returned (only metadata + last-4 prefix).

Run:
    CORELINK_PAT=$PAT python3 examples/quickstart_list.py
"""

from __future__ import annotations

import json
import os
import sys
import urllib.error
import urllib.request


def main() -> int:
    api = os.environ.get("CORELINK_API_URL", "https://sandbox.corelink.humangr.com")
    pat = os.environ.get("CORELINK_PAT")
    if not pat:
        print("error: CORELINK_PAT env var is required", file=sys.stderr)
        return 2

    req = urllib.request.Request(
        url=f"{api}/v1/pats",
        method="GET",
        headers={"Authorization": f"Bearer {pat}", "Accept": "application/json"},
    )
    try:
        with urllib.request.urlopen(req, timeout=10) as resp:
            payload = json.loads(resp.read().decode("utf-8"))
            items = payload.get("items") if isinstance(payload, dict) else None
            count = len(items) if isinstance(items, list) else 0
            print(f"status={resp.status} pat_count={count}")
            print(json.dumps(payload, indent=2))
    except urllib.error.HTTPError as exc:
        print(f"http_error status={exc.code} body={exc.read().decode('utf-8', 'replace')}", file=sys.stderr)
        return 1
    except urllib.error.URLError as exc:
        print(f"network_error: {exc.reason}", file=sys.stderr)
        return 1
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
